pydantic_source/source.py

```diff
@@ -0,0 +1,158 @@
+import json
+import ast
+from typing import Any, Type, Dict
+from benedict import benedict
+from munch import Munch
+
+import yaml
+from pydantic_settings import BaseSettings, PydanticBaseSettingsSource
+from pydantic.fields import FieldInfo
+
+from rapyuta_io_sdk_v2 import Client, Configuration
+import base64
+
+
+class ConfigTreeSource(PydanticBaseSettingsSource):
+    def __init__(
+        self,
+        settings_cls: Type["BaseSettings"],
+        config: Configuration,
+        tree_name: str = "default",
+        key_prefix: str = "",
+        local_file: str = None,
+    ):
+        super().__init__(settings_cls)
+        self.client = Client(config=config)
+        self.tree_name = tree_name
+        self.local_file = local_file
```

Comment:
suggestion: Consider making all the fields and methods private.

```suggestion
        self._client = Client(config=config)
        self._tree_name = tree_name
        self._local_file = local_file
```

Reply:
Some fields and methods are still public.